src/tables.jl

```diff
@@ -132,91 +146,101 @@ column name `:Ti`, and `Dim{:custom}` becomes `:custom`.
 To get dimension columns, you can index with `Dimension` (`X()`) or
 `Dimension` type (`X`) as well as the regular `Int` or `Symbol`.
 """
-struct DimTable{Keys,S,DC,DAC} <: AbstractDimTable
-    stack::S
-    dimcolumns::DC
-    dimarraycolumns::DAC
-end
-DimTable{K}(stack::S, dimcolumns::DC, strides::SD) where {K,S,DC,SD} = 
-    DimTable{K,S,DC,SD}(stack, dimcolumns, strides)
-DimTable(A::AbstractDimArray, As::AbstractDimArray...) = DimTable((A, As...))
-function DimTable(As::Tuple{AbstractDimArray,Vararg{AbstractDimArray}}...)
-    DimTable(DimStack(As...))
-end
-function DimTable(s::AbstractDimStack)
+struct DimTable <: AbstractDimTable
+    colnames::Vector{Symbol}
+    dimcolumns::Vector{DimColumn}
+    dimarraycolumns::Vector{DimArrayColumn}
+end
+
+function DimTable(s::AbstractDimStack; mergedims=nothing)
+    s = isnothing(mergedims) ? s : DimensionalData.mergedims(s, mergedims)
     dims_ = dims(s)
     dimcolumns = map(d -> DimColumn(d, dims_), dims_)
     dimarraycolumns = map(A -> DimArrayColumn(A, dims_), s)
     keys = _colnames(s)
-    DimTable{keys}(s, dimcolumns, dimarraycolumns)
+    return DimTable(collect(keys), collect(dimcolumns), collect(dimarraycolumns))
+end
+
+function DimTable(xs::Vararg{AbstractDimArray}; layernames=[Symbol("layer_$i") for i in eachindex(xs)], mergedims=nothing)
+    # Check that dims are compatible
+    comparedims(xs...)
+
+    # Construct DimColumns
+    xs = isnothing(mergedims) ? xs : map(x -> DimensionalData.mergedims(x, mergedims), xs)
+    dims_ = dims(first(xs))
+    dimcolumns = collect(map(d -> DimColumn(d, dims_), dims_))
+    dimnames = collect(map(dim2key, dims_))
+
+    # Construct DimArrayColumns
+    dimarraycolumns = collect(map(A -> DimArrayColumn(A, dims_), xs))
+
+    # Return DimTable
+    colnames = vcat(dimnames, layernames)
+    return DimTable(colnames, dimcolumns, dimarraycolumns)
+end
+
+function DimTable(x::AbstractDimArray; layersfrom=nothing, mergedims=nothing)
+    if !isnothing(layersfrom) && any(hasdim(x, layersfrom))
+        nlayers = size(x, layersfrom)
+        layers = [(@view x[layersfrom(i)]) for i in 1:nlayers]
+        layernames = Symbol.(["$(dim2key(layersfrom))_$i" for i in 1:nlayers])
+        return DimTable(layers..., layernames=layernames, mergedims=mergedims)
+    else
+        return DimTable(DimStack((;value=x)), mergedims=mergedims)
+    end
 end
 
 dimcolumns(t::DimTable) = getfield(t, :dimcolumns)
 dimarraycolumns(t::DimTable) = getfield(t, :dimarraycolumns)
+colnames(t::DimTable) = getfield(t, :colnames)
 dims(t::DimTable) = dims(parent(t))
 
-Base.parent(t::DimTable) = getfield(t, :stack)
+Base.parent(t::DimTable) = getfield(t, :colnames)
 
 for func in (:dims, :val, :index, :lookup, :metadata, :order, :sampling, :span, :bounds,
              :locus, :name, :label, :units)
     @eval $func(t::DimTable, args...) = $func(parent(t), args...)
 
```

Comment:
What do we want to do about parent? We are no longer storing the source array/stack inside DimTable because doing so might require converting a DimArray to a DimStack with a large number of layers, which is slow. However, that means these methods will be broken. Could we perhaps just store the first layer instead of the full stack?

Reply:
Right. Long-term we do really need to add that wide `DimStack` so that's not a problem

But there is a pending minor version bump anyway, so we can remove `parent` if its not needed.